### PR TITLE
Fix `Base.mapreduce` to match the new signature in 0.7/1.0

### PR DESCRIPTION
**Pull request by ssz66666**
In Julia 0.7/1.0, the signature of `Base.mapreduce` and its alternative function `Base.mapreducedim` which allows a `dims` argument, have [changed](https://github.com/JuliaLang/julia/pull/27711) to use keyword arguments, i.e.
from
`mapreduce(f, op[, v0], itr)`
`mapreducedim(f, op, itr, region[, v0])`
to
`mapreduce(f, op, itr; dims=:, [init])`

It seems that the bot wasn't able to figure out that this is deprecated syntax. The current head is still passing tests since no test case tries to pass a customised initial value to it.

### CODE REVIEW

**Comment by vchuravy** [comment]
Thanks! It would be greatly appreciated if you can add testcases for this as well!

File: src/abstractarray.jl [hunk 1]
```diff
@@ -56,14 +56,24 @@ end
...
-Base.show(io::IO, x::LinearAlgebra.Adjoint{<:Any,<:GPUArray}) =
-    Base.show(io, LinearAlgebra.adjoint(Array(x.parent)))
-Base.show(io::IO, x::LinearAlgebra.Transpose{<:Any,<:GPUArray}) =
-    Base.show(io, LinearAlgebra.transpose(Array(x.parent)))
-
-Base.show_vector(io::IO, x::GPUArray) = Base.show_vector(io, Array(x))
-
+for (atype, op) in
+    [(:(GPUArray), :(Array)),
+     (:(LinearAlgebra.Adjoint{<:Any,<:GPUArray}), :(x->LinearAlgebra.adjoint(Array(x.parent)))),
```
Comment (maleadt) [inline src/abstractarray.jl]: maybe use `Base.parent` instead of accessing the field directly

File: src/testsuite.jl [hunk 2]
```diff
@@ -49,6 +49,7 @@ end
 Runs the entire GPUArrays test suite on array type `AT`
 """
 function test(AT::Type{<:GPUArray})
+    GPUArrays.allowscalar(false)
```
Comment (maleadt) [inline src/testsuite.jl]: my reasoning here was that implementations can decide themselves if they want to assert scalar indexing, eg. https://github.com/JuliaGPU/CuArrays.jl/blob/master/test/runtests.jl#L27
but then it probably needs to be added to the JLArray tests.

Comment (ssz66666) [reply src/testsuite.jl]: Then maybe moving scalar indexing assertion into runtest.jl of GPUArrays package for JLArray tests, like
```julia
@testset "JLArray" begin
    local _prev = GPUArrays._allowscalar[]
    GPUArrays.allowscalar(false)
    GPUArrays.test(JLArray)
    GPUArrays._allowscalar[] = _prev
end
```

Comment (maleadt) [reply src/testsuite.jl]: Something like that, yeah. But we can leave that for a different PR.

**Comment by ssz66666** [comment]
Fixed wrapper field access, will submit a separate PR for the test suite behaviour change.

File: src/testsuite.jl (hunk 2 above)
Comment (maleadt) [reply src/testsuite.jl]: (something like `@allowscalar`, eg. `@disallowscalar` would be cleaner then, not having to work with unexported values in GPUArrays)

**Comment by maleadt** [comment]
Thanks! Too bad we can't run CI on this, guess we'll have to keep a close eye on master after merging PRs like this.

**Comment by maleadt** [comment]
OK, so this apparently broke CuArrays (as per CI), due to passing a CPU array to the GPU. Once https://github.com/JuliaLang/METADATA.jl/pull/17380 is merged CI will report better errors.

**Comment by ssz66666** [comment]
Hmm, don't know why the added unit test broke CuArrays. CUDAnative fails to compile the GPU kernel for some reason. Right now it seems that it is the occurrence of anonymous function `_addone` that is causing the problem. manually running the code in Main outside the testsuite seemed to work fine...

**Pull request merged**